extensions/amazon-aws/src/glue.tsx

```diff
@@ -0,0 +1,291 @@
+import {
+  GlueClient,
+  ListJobsCommand,
+  GetJobRunsCommand,
+  JobRun,
+  GetJobCommand,
+  StartJobRunCommand,
+  GetJobCommandOutput,
+} from "@aws-sdk/client-glue";
+import { Action, ActionPanel, Icon, List, Image, Color, Detail, showToast, Toast } from "@raycast/api";
+import { MutatePromise, showFailureToast, useCachedPromise } from "@raycast/utils";
+import AWSProfileDropdown from "./components/searchbar/aws-profile-dropdown";
+import { isReadyToFetch, resourceToConsoleLink } from "./util";
+import { AwsAction } from "./components/common/action";
+
+interface GlueJobRun extends JobRun {
+  accessoriesText?: string;
+  icon?: Icon;
+  iconTintColor?: Color;
+}
+
+export default function Glue() {
+  const { data: glueJobs, error, isLoading, mutate } = useCachedPromise(fetchJobs);
+  const {
+    data: glueJobRuns,
+    isLoading: isLoadingJobRuns,
+    mutate: mutateJobRuns,
+  } = useCachedPromise(fetchJobRuns, [1, glueJobs]);
+
+  return (
+    <List
+      isLoading={isLoading || isLoadingJobRuns}
+      searchBarPlaceholder="Filter Jobs by name..."
+      searchBarAccessory={<AWSProfileDropdown onProfileSelected={mutate} />}
+      isShowingDetail
+    >
+      if (error)
+      {error ? (
+        <List.EmptyView title={error.name} description={error.message} icon={Icon.Warning} />
+      ) : (
+        glueJobRuns?.map((job) => <GlueJob job={job} mutate={mutateJobRuns} />)
+      )}
+    </List>
+  );
+}
+
+function GlueJob({ job: glueJobRun, mutate }: { job: GlueJobRun; mutate: MutatePromise<GlueJobRun[] | undefined> }) {
+  return (
+    <List.Item
+      key={glueJobRun.JobName}
+      icon={{ source: "aws-icons/glue.png", mask: Image.Mask.RoundedRectangle }}
+      title={glueJobRun.JobName || ""}
+      detail={<GlueJobRunDetails jobRun={glueJobRun} />}
+      actions={
+        <ActionPanel>
+          <Action.Push
+            title="List all Job Runs"
+            icon={Icon.List}
+            target={<GlueJobRuns glueJobName={glueJobRun.JobName!} />}
+          />
+          <Action title="Start Run" onAction={() => RunGlueJob(glueJobRun.JobName!, mutate)} icon={Icon.Play} />
+          <Action.Push
+            title="Show Job Definition"
+            icon={Icon.Info}
+            target={<GlueJobDefinition glueJobName={glueJobRun.JobName!} />}
+            shortcut={{ modifiers: ["cmd"], key: "d" }}
+          />
+          <Action
+            title="Refresh"
+            onAction={() =>
+              mutate(fetchJobRuns(1, [glueJobRun.JobName!]), {
+                optimisticUpdate(data) {
+                  return data;
+                },
+              })
+            }
+            icon={Icon.RotateAntiClockwise}
+          />
```

Comment:
Ahh this is a little more complicated. I see you want to just get the latest state for the job and ideally wouldn't want to revalidate all jobs. For this case, what you had before and my recommendation do not work.

I think to keep things simple here, we just need:
1. a call to fetch the latest job run for the given job, say we store it in `newJobRun = (await fetchJobRuns(1, [glueJobRun.JobName!]))[0]`
2. we just filter and override the job state in the `glueJobRuns` list we are using throughout, so `glueJobRuns = glueJobRuns.map(j => j.Id !== glueJobRun.Id ? j : {...j, ...newJobRun} as GlueJobRun)`

I think this would achieve what you want w/o requiring a full-stack mutation. If this doesn't work try to do this with mutation, with `shouldRevalidateAfter` set to `false`. What do you think?